system/hiera.py

```diff
@@ -0,0 +1,176 @@
+        required: true
+        default: null
+        aliases: ['name']
+    context:
+        description:
+            - This key value will set the scope of hiera key/value. Also will follow the hiera's hierarchi, then if you dont have any variable in the first scope will go down to the next one.
+        required: false
+        default: null
+    source:
+        description:
+            - The hiera config file path, if you want to custom every query with multiple hierarchies
+        required: false
+        default: null
+'''
+
+EXAMPLES = '''
+# Without Puppetmaster example, stand-alone node
+---
+- name: Test
+  hosts: 127.0.0.1
+  tasks:
+    - name: Retrieving Hiera Data
+      hiera: path=/bin/hiera key="{{ item.value }}" fact="{{ item.key }}" source=/etc/hiera.yaml
+      args:
+        context:
+          environment: 'production'
+          fqdn: 'puppet01.localdomain'
+      with_dict:
+        var_array_multi: "proxy::array_multi"
+        var_array_line: "proxy::array_line"
+        line: "line"
+
+    - debug: msg="{{ item }}"
+      with_items: var_array_multi
+    - debug: msg="{{ item }}"
+      with_items: var_array_line
+    - debug: msg="{{ line }}"
+
+# Puppetmaster example
+---
+- name: Create Facts with Hiera Data
+  hosts: nodes
+  sudo: yes
+  tasks:
+    - name: Copy Hiera Data into Destination node
+      copy: src=/var/lib/hiera/ dest=/var/lib/
+
+    - name: Copy Hiera Config into Destination node
+      copy: src=/etc/hiera.yaml dest=/etc/hiera.yaml
+
+    - name: Retrieving Hiera Data
+      hiera: path=/bin/hiera key="{{ item.value }}" fact="{{ item.key }}" source=/etc/hiera.yaml
+      args:
+        context:
+          environment: 'production'
+          fqdn: 'puppet01.localdomain'
+      with_dict:
+        var_array_multi: "proxy::array_multi"
+        var_array_line: "proxy::array_line"
+        line: "line"
+
+    - debug: msg="{{ item }}"
+      with_items: var_array_multi
+    - debug: msg="{{ item }}"
+      with_items: var_array_line
+    - debug: msg="{{ line }}"
+'''
+
+RETURN = '''
+This module will return the datatype that you got in Hieradata backend.
+
+'''
+
+
+def main():
+    """
+    Main function.
+    This module will parse your hiera hierarchi and will return the required
+    values
+    - key: Is the key name of the hiera variable
+    - fact: Is the key name that must store the hiera output
+    - args: context: must contain all the values that identify the node against hiera
+    - path: hiera executable
+    - source: hiera config file
+        WARNING: This module try to solve a disparity between arch of
+        Puppet/Hiera and Ansible, because of how they works (Puppet compile
+        the values into puppet master node) (Ansible are executed into
+        destination node and have not access to Hiera backend directly)
+
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(aliases=['key']),
```

Comment:
here you have `name` as the primary and `key` as the alias, in `DOCUMENTATION` it's the other way around. Also, you have `required: True` in `DOCUMENTATION` and here it's `False` (as that's the default value), should have `required=True`